### ISSUE TITLE
[TS] Error when specifying destination type

### ISSUE THREAD

**Issue by stefanosandes** [nobrainr/morphism#121]
**Describe the bug**
I'm trying to get a typed  array, but typescript is warning when I specifying the destination type class on `morphism(schema, source, Destination)`.

**Screenshots**
<img width="810" alt="CleanShot 2019-08-27 at 13 06 12@2x" src="https://user-images.githubusercontent.com/6722075/63788371-bc06f680-c8cb-11e9-8348-425b15a6d83d.png">

The error description:
```
Argument of type 'Destination[]' is not assignable to parameter of type 'Source'.
  Type 'Destination[]' is missing the following properties from type 'Source': id, ugly_fieldts(2345)
```
**Additional context**
Environment: Nodejs
Ts version: 3.5.3

Screenshot code:

```typescript
import { StrictSchema, morphism } from 'morphism';

class Source {
  public id: number;
  public ugly_field: string;
}

class Destination {
  constructor(public id: number, public field: string) {}
}

const source = [new Destination(1, 'abc'), new Destination(1, 'def')];

const schema: StrictSchema<Destination, Source> = {
  id: 'id',
  field: 'ugly_field',
};

const result = morphism(schema, source, Destination);
```

**Comment by emyann** [comment]
@stefanosandes This typescript issue has been fixed with the latest release: [`Morphism@1.12.1`](https://github.com/nobrainr/morphism/releases/tag/v1.12.1)

You can take a look at this repl to see it in action: https://repl.it/@yrnd1/EqualForsakenDisk

Please let me know if you encounter any issue of this type again :)

**Comment by stefanosandes** [comment]
@emyann awesome!! Thanks for your attention to this issue! And thanks for this great support and the excellent job in this library.
